In [93]:
import pandas as pd
import numpy as np
import seaborn as sn
import matplotlib.pyplot as plyt
%matplotlib inline
from scipy import stats
from scipy.stats import pearsonr

In [94]:
df = pd.read_csv(r"C:\Users\ADMIN\Desktop\DS\1. Weather Data.csv")

Data Overview and Cleaning

In [95]:
df.head


<bound method NDFrame.head of              Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0        1/1/2012 0:00    -1.8              -3.9         86                4   
1        1/1/2012 1:00    -1.8              -3.7         87                4   
2        1/1/2012 2:00    -1.8              -3.4         89                7   
3        1/1/2012 3:00    -1.5              -3.2         88                6   
4        1/1/2012 4:00    -1.5              -3.3         88                7   
...                ...     ...               ...        ...              ...   
8779  12/31/2012 19:00     0.1              -2.7         81               30   
8780  12/31/2012 20:00     0.2              -2.4         83               24   
8781  12/31/2012 21:00    -0.5              -1.5         93               28   
8782  12/31/2012 22:00    -0.2              -1.8         89               28   
8783  12/31/2012 23:00     0.0              -2.1         86               30   

      Vis

In [96]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8784 entries, 0 to 8783
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Date/Time         8784 non-null   object 
 1   Temp_C            8784 non-null   float64
 2   Dew Point Temp_C  8784 non-null   float64
 3   Rel Hum_%         8784 non-null   int64  
 4   Wind Speed_km/h   8784 non-null   int64  
 5   Visibility_km     8784 non-null   float64
 6   Press_kPa         8784 non-null   float64
 7   Weather           8784 non-null   object 
dtypes: float64(4), int64(2), object(2)
memory usage: 549.1+ KB


The data set provides various weather parameters and the respective dates/times these parameters were measured. The measured parameters include temperature, humidity, wind speed, visibility and pressure. 
The total nmber of entries in the data set are 8784 entries of data types integers floats and objects.

In [97]:
check_nulls = df.isnull().any()
check_nulls

Date/Time           False
Temp_C              False
Dew Point Temp_C    False
Rel Hum_%           False
Wind Speed_km/h     False
Visibility_km       False
Press_kPa           False
Weather             False
dtype: bool

There are no null values in the entire data set

In [98]:
check_duplicates = df.duplicated().any()
check_duplicates

False

There are no duplicate values in the dataset

STATISTICAL OVERVIEW

In [99]:
df = df.describe()
df

Temp_C  Dew Point Temp_C    Rel Hum_%  Wind Speed_km/h  \
count  8784.000000       8784.000000  8784.000000      8784.000000   
mean      8.798144          2.555294    67.431694        14.945469   
std      11.687883         10.883072    16.918881         8.688696   
min     -23.300000        -28.500000    18.000000         0.000000   
25%       0.100000         -5.900000    56.000000         9.000000   
50%       9.300000          3.300000    68.000000        13.000000   
75%      18.800000         11.800000    81.000000        20.000000   
max      33.000000         24.400000   100.000000        83.000000   

       Visibility_km    Press_kPa  
count    8784.000000  8784.000000  
mean       27.664447   101.051623  
std        12.622688     0.844005  
min         0.200000    97.520000  
25%        24.100000   100.560000  
50%        25.000000   101.070000  
75%        25.000000   101.590000  
max        48.300000   103.650000

The overview privides 8 counts of data points. The mean values, max&min values,standard deviation and the respective percentiles for each of the columns are provied.

In [100]:
Q1 = df.quantile(0.25)
Q3 = df.quantile(0.75)
IQR = Q3 - Q1
Q3

Temp_C               22.350000
Dew Point Temp_C     14.950000
Rel Hum_%            85.750000
Wind Speed_km/h      35.750000
Visibility_km        32.823335
Press_kPa           102.105000
Name: 0.75, dtype: float64

In [101]:
outliers = df[((df < (Q1 - 1.5 * IQR)) | (df > (Q3 + 1.5 * IQR))).any(axis = 1)]


DATA VISUALIZATION